packages/lesswrong/components/users/account/UsersEditForm.tsx

```diff
@@ -43,35 +35,22 @@ const passwordResetMutation = gql`
 `
 
 const UsersEditForm = ({terms, classes}: {
-  terms: {slug?: string, documentId?: string},
+  terms: {slug: string},
   classes: ClassesType,
 }) => {
   const currentUser = useCurrentUser();
   const { flash } = useMessages();
   const navigate = useNavigate();
   const client = useApolloClient();
-  const { Typography } = Components;
+  const { ErrorAccessDenied } = Components;
   const [ mutate, loading ] = useMutation(passwordResetMutation, { errorPolicy: 'all' })
   const currentThemeOptions = useThemeOptions();
   const setTheme = useSetTheme();
 
-  if(!terms.slug && !terms.documentId) {
-    // No user specified and not logged in
-    return (
-      <div className={classes.root}>
-        Log in to edit your profile.
-      </div>
-    );
-  }
-  if (!userCanEditUser(currentUser,
-    terms.documentId ?
-      {_id: terms.documentId} :
-      // HasSlugType wants some fields we don't have (schemaVersion, _id), but
-      // userCanEdit won't use them
-      {slug: terms.slug, __collectionName: 'Users'} as HasSlugType
-  )) {
-    return <span>Sorry, you do not have permission to do this at this time.</span>
```

Comment:
I combined these two cases by always using `slug`. I thought there might be some reason to use `documentId`, such as:
- The currentUser fragment doesn't get cleared from the cache when updating
- There is some weirdness when editing your own slug

I checked both of these and didn't find an issue, so I'm pretty sure this is fine to do